solidity/contracts/relay/LightRelayMaintainerProxy.sol

```diff
@@ -0,0 +1,128 @@
+// SPDX-License-Identifier: GPL-3.0-only
+
+// ██████████████     ▐████▌     ██████████████
+// ██████████████     ▐████▌     ██████████████
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+// ██████████████     ▐████▌     ██████████████
+// ██████████████     ▐████▌     ██████████████
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+
+pragma solidity 0.8.17;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@keep-network/random-beacon/contracts/Reimbursable.sol";
+import "@keep-network/random-beacon/contracts/ReimbursementPool.sol";
+
+import "./LightRelay.sol";
+
+/// @title LightRelayMaintainerProxy
+/// @notice The proxy contract that allows the relay maintainers to be refunded
+///         for the spent gas from the `ReimbursementPool`. When proving the
+///         next Bitcoin difficulty epoch, the maintainer calls the
+///         `LightRelayMaintainerProxy` which in turn calls the actual `LightRelay`
+///         contract.
+contract LightRelayMaintainerProxy is Ownable, Reimbursable {
+    ILightRelay public lightRelay;
+
+    /// @notice Stores the addresses that can maintain the relay. Those
+    ///         addresses are attested by the DAO.
+    /// @dev The goal is to prevent a griefing attack by frontrunning relay
+    ///      maintainer which is responsible for retargetting the relay in the
+    ///      given round. The maintainer's transaction would revert with no gas
+    ///      refund. Having the ability to restrict maintainer addresses is also
+    ///      important in case the underlying relay contract has authorization
+    ///      requirements for callers.
+    mapping(address => bool) public isAuthorized;
+
+    /// @notice Gas that is meant to balance the retarget overall cost. Can be
+    //          updated by the governance based on the current market conditions.
+    uint256 public retargetGasOffset;
+
+    event LightRelayUpdated(address newRelay);
+
+    event MaintainerAuthorized(address indexed maintainer);
+
+    event MaintainerDeauthorized(address indexed maintainer);
+
+    event RetargetGasOffsetUpdated(uint256 retargetGasOffset);
+
+    modifier onlyRelayMaintainer() {
+        require(isAuthorized[msg.sender], "Caller is not authorized");
+        _;
+    }
+
+    modifier onlyReimbursableAdmin() override {
+        require(owner() == msg.sender, "Caller is not the owner");
+        _;
+    }
+
+    constructor(ILightRelay _lightRelay, ReimbursementPool _reimbursementPool) {
+        lightRelay = _lightRelay;
+        reimbursementPool = _reimbursementPool;
+
+        retargetGasOffset = 52000;
```

Comment:
This value should depend on the number of headers we use to `retarget` the `LightRelay`.
If we trust unit tests then this value should be between `50000` and `54000` for the proof length of `20` (which is the value we currently have in `LightRelay`).
If it's below `50000` the caller ends up with less ETH than before the call.
If it's above `54000` the caller ends up with more than `0.001` ETH.

Reply:
Let's do 54000. It's better to have the maintainer slightly net-positive than net-negative. 

Note taken on revisiting this value. Let's keep monitoring it after the mainnet deployment.

Reply:
Done in 73d256987d73856dc0a127b808678599bbbe26ed.